### PR TITLE
Add regression tests

### LINKED ISSUE #417: Inconsistencies between buf and protoc
These are some inconsistencies between buf and protoc found in #411

### Form Feeds and Vertical Tabs

Protobufs containing the characters "\f" and "\v" fail with `buf generate` but succeeds with protoc. Buf outputs the message "syntax error: unexpected $unk". This seems to be coming from [this line](https://github.com/jhump/protoreflect/blob/fb56a945d0735c55f3bb1853ad1f99766ac126e7/desc/protoparse/proto.y.go#L925) in "github.com/jhump/protoreflect". I'm not familiar with goyacc, so my diagnosis stops there.

### Duplicate message and rpc names

Buf allows an rpc's name to be the same as a message's name in the same protobuf, but protoc does not.

For example:

```protobuf
syntax="proto3";

package a;

option go_package = "foo.bar/foo";

message Foo{}
message Bar{}

service MyService{
  rpc Foo(Foo) returns (Bar);
}
```

Buf generates code for ☝️, but protoc errors with the message: `"Foo" is not a message type.`

### Space after first number in extensions

With this protobuf:

```protobuf
syntax="proto2";

package a;

option go_package = "foo.bar/foo";

message Test{
  extensions 10to max;
}
```

protoc fails with `Need space between number and identifier.` but Buf generates code.

### PR DESCRIPTION
**Pull request by WillAbides**
This adds some regression tests for issues fixed in #430

### CODE REVIEW

**Comment by bufdev** [comment]
We have this covered internally, will copy over!

**Comment by bufdev** [comment]
This is mostly supplanted by https://github.com/bufbuild/buf/pull/430 - you don't have to, but any chance you could merge that in/revert your common changes here?

File: internal/buf/bufimage/bufimagebuild/builder_test.go [hunk 1]
```diff
@@ -314,3 +331,12 @@ func testGetImageImportPaths(image bufimage.Image) []string {
 	sort.Strings(importNames)
 	return importNames
 }
+
+func testFileAnnotations(t *testing.T, relDirPath string, wantMessages ...string) {
```
Comment (WillAbides) [inline internal/buf/bufimage/bufimagebuild/builder_test.go:335]: This hit the 3 times threshold so I extracted it.

**Comment by WillAbides** [comment]
I merged main and updated the PR to reflect it's new purpose.

File: internal/buf/bufimage/bufimagebuild/builder_test.go (hunk 1 above)
Comment (bufdev) [reply internal/buf/bufimage/bufimagebuild/builder_test.go:335]: We don't have a 3 times threshold, but this is fine

**Review by bufdev — APPROVED** [review]

**Pull request merged**